src/Microsoft.DotNet.Build.Tasks/PackageFiles/FrameworkTargeting.targets

```diff
@@ -102,6 +102,26 @@
     </PropertyGroup>
   </Target>
 
+  <!--
+    Cross Platform MSBuild has some logic to replace \ with / when invoking commands to fix up path differences between Windows and
+    *NIX. The define command line argument syntax for VB requires that we both surround some items with quotes and escape the quotes with
+    backslashes. However, due to the above MSBuild logic, this causes an invalid command line to be generated when running on *NIX.
+
+    Microsoft/msbuild#422 tracks an actual fix in MSBuild, but for now we work around the issue by using a custom task that
+    transforms the set of defines we are going to use into a response file we can pass along to the Vbc task along with an
+    empty set of defines.
+  -->
+  <UsingTask TaskName="WriteVisualBasicDefineResponseFile"  AssemblyFile="$(BuildToolsTaskDir)Microsoft.DotNet.Build.Tasks.dll" />
+
+  <Target Name="ConvertDefinesToResonseFile" BeforeTargets="CoreCompile" Condition="'$(MSBuildProjectExtension)' == '.vbproj'">
+    <WriteVisualBasicDefineResponseFile DefineConstants="$(FinalDefineConstants)"
```

Comment:
I tried using `WriteLinesToFile` with some transformations so we didn't need this task, but slashes were still getting converted in that case.